lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -131,6 +136,16 @@ def supports_network_tokenization?
 
       private
 
+      def add_field_if_present(post, options, field)
+        post[field] = options[field] if options[field]
+      end
```

Comment:
👍I feel like a great portion of Active Merchant is testing for the presence of a field and adding it if present. Seems like there should be a base method like this that also has an option default to use if the field is not present.

Reply:
FYI Benjamin Pollock did some exploratory work along these lines, but we never revisited it: https://github.com/activemerchant/active_merchant/pull/3067

Reply:
In general though, it's much more common for adapters to check for presence in the line which adds the element. I think this is generally because there could be some mapping to different field names and seeing it all in one place is a bit more readable (from a certain perspective)

Reply:
That said, I have nothing against this implementation in theory, it just feels a little wierd for it to be only used for a small subset of field additions.

Reply:
I streamlined the helper further and moved it to the base class with tests. That way, we can employ it in the future in any gateway implementation, and extend with other options if desired. Grepping the code, I saw this pattern hundreds of times, so I tend to think it's worthwhile to have a helper to start DRYing things out, even if exceptions like mapping to a different field name can't use it. Let me know if there are any strong preferences to the contrary though.

Reply:
I think I like this route. It might be best to keep it simple, forcing the exceptions to be explicit (though it might be best to add some common exceptions, sparingly). The challenge will be remembering to use it 😄  
I do have a couple small suggestions for the base methods.

Reply:
👍 Should the base methods be named in such a way to indicate it will only work for request params/JSON API's rather than XML?

Small nitpick on the method arguments, I'd prefer more descriptive names for the method args or at least comments to indicate what they represent.

Reply:
Good point. It might be worth making it handle both JSON and XML.

Reply:
Good point on the naming. Since the context is the gateway class instead of somewhere like ```Hash```, naming according to the intended use does work better. Let me know if anything is still not clear. I'd prefer to add the functionality for XML (which might be in a different helper altogether) later at some point when it'd be used, for the sake of scope and testing.

Reply:
Agreed. We have already expanded scope on the original change so this makes sense.